pandas/core/arrays/datetimelike.py

```diff
@@ -253,9 +253,24 @@ def _check_compatible_with(
     # ------------------------------------------------------------------
     # NDArrayBackedExtensionArray compat
 
+    def __setstate__(self, state):
+        if isinstance(state, dict):
```

Comment:
woa, but ok. can we create an issue to track *when* we should remove things like this (e.g. call it pickle compatibility and point to this PR). will be a while for sure, but should track.

Reply:
sure.  yah, pickle backwards compat is a PITA